src/signals-unix.c

```diff
@@ -525,11 +525,14 @@ void usr2_handler(int sig, siginfo_t *info, void *ctx)
     jl_atomic_exchange(&ptls->signal_request, 0); // returns -1
     if (request == 2) {
         int force = jl_check_force_sigint();
+        if (!force && want_interrupt_handlers()) {
+            return;
+        }
```

Comment:
Seems like we should try to decide whether this is the responsibility of sigusr2_handler or signal_listener thread. Seems like the signal thread usually handles it on all of the other OS.
```suggestion
```

Reply:
Makes sense, removed